src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -45,37 +43,38 @@ export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundarySt
     window.removeEventListener("unhandledrejection", this.onUnhandledRejection);
   }
 
+  /**
+   * Global promise unhandled promise rejection to trigger error boundary
+   */
   onUnhandledRejection = (event: PromiseRejectionEvent): void => {
-    const { onError } = this.props;
     event.preventDefault();
     event.promise.catch(async (error) => {
-      stack(error);
-      this.setState(ErrorBoundary.getDerivedStateFromError(error), () => {
-        if (onError) onError(error);
-      });
+      this.setState(ErrorBoundary.getDerivedStateFromError(error));
     });
   };
 
+  /**
+   * Attempts to recover from error.
+   */
   recover = (): void => {
     const { onRecover } = this.props;
-    this.setState(
-      {
-        hasError: false,
-        error: undefined,
-      },
-      () => {
-        if (onRecover) onRecover();
-      }
-    );
+    onRecover();
+
+    this.setState({
+      hasError: false,
+      error: undefined,
+    });
   };
 
   render(): ReactNode {
-    const error = this.state.error;
-    const {
-      recover,
-      props: { renderer: ErrorRenderer },
-    } = this;
+    const { recover } = this;
+    const { hasError, error } = this.state;
+    const { children, FallbackComponent } = this.props;
+
+    if (hasError) {
+      return <FallbackComponent error={error} recover={recover} />;
+    }
 
-    return this.state.hasError ? <ErrorRenderer error={error} recover={recover} /> : this.props.children;
+    return children;
```

Comment:
Thanks for cleaning up this part! Just realise I destructured some variables and yet not all from the state and props previously. Lol!🙇‍♂️